src/libraries/System.Numerics.Tensors/src/System/ThrowHelper.cs

```diff
@@ -57,13 +74,19 @@ internal static void ThrowIndexOutOfRangeException()
         }
 
         [DoesNotReturn]
-        public static void ThrowArgumentException_DestinationTooShort()
+        public static void ThrowArgument_LengthsMustEqualArrayLength()
         {
-            throw GetArgumentException(SR.DestinationTooShort);
+            throw new ArgumentOutOfRangeException();
```

Comment:
> throw new ArgumentOutOfRangeException(); [](http://example.com/codeflow?start=11&length=41)

nit: shouldn't such exceptions mention the argument name?

Reply:
We have a range that do and don't. Exceptions need some additional cleanup that we shouldn't block the next preview on, particularly since they typically represent developer error rather than something you'd need dynamically at runtime.